include/bm/bm_sim/field_lists.h

```diff
@@ -46,59 +51,86 @@ class FieldList {
     bool operator==(const field_t& other) const {
       return header == other.header && offset == other.offset;
     }
-
     bool operator!=(const field_t& other) const {
       return !(*this == other);
     }
   };
 
- public:
-  using iterator = std::vector<field_t>::iterator;
-  using const_iterator = std::vector<field_t>::const_iterator;
-  using reference = std::vector<field_t>::reference;
-  using const_reference = std::vector<field_t>::const_reference;
-  using size_type = size_t;
+  struct constant_t {
+    ByteContainer value;
+
```

Comment:
I guess it's possible that we'll have a constant as part of a field list that has some not-divisible-by-eight bitwidth and want to perform a calculation on the field list as a contiguous buffer. I'll add it back.